server/store.ml

```diff
@@ -53,7 +53,7 @@ module Node = struct
 
   let replace_child node child nchild =
     (* this is the on-steroid version of the filter one-replace one *)
-    let rec replace_one_in_list l =
+    let[@tail_mod_cons] rec replace_one_in_list l =
```

Comment:
This has is recognised from OCaml 4.14 onwards, for older versions of the compiler it's ignored, so it shouldn't cause problems

Reply:
I don't have older compiler versions at hand to check, but godbolt confirms that the attribute is ignored by OCaml 4.04 for example: https://godbolt.org/z/zKxxGoafY